storage/remote/client_test.go

```diff
@@ -0,0 +1,71 @@
+package remote
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+)
+
+func TestStoreHTTPErrorHandling(t *testing.T) {
+	tests := []struct {
+		code        int
+		shouldFail  bool
+		recoverable bool
+	}{
+		{
+			code:       200,
+			shouldFail: false,
+		},
+		{
+			code:        300,
+			shouldFail:  true,
+			recoverable: false,
+		},
+		{
+			code:        404,
+			shouldFail:  true,
+			recoverable: false,
+		},
+		{
+			code:        500,
+			shouldFail:  true,
+			recoverable: true,
+		},
+	}
+
+	for i, test := range tests {
+		server := httptest.NewServer(
+			http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				http.Error(w, "test error", test.code)
+			}),
+		)
+
+		serverURL, err := url.Parse(server.URL)
+		if err != nil {
+			panic(err)
```

Comment:
I'd keep this a panic, as it should never fail, as it would be a bug in the test code.